### DISCUSSION TITLE
Not able to DeDRM an epub

### DISCUSSION THREAD

**Discussion by irisvoo** [apprenticeharper/DeDRM_tools#1858] [Q&A]
Hi everyone, 

I am trying to DeDRM my ebook on Windows 10 but it just won't work, not in the latest versions of Calibre &DeDRM and also not in the 4.23 and 6.8.1 versions (mentioned in #1692 ). I purchased the book via a website that worked together with my Kobo Glo HD so when I logged into my Kobo it synchronizes my purchased books automatically. However, when I want to read it on another device, the website says I need to download it, it becomes an .ascm format, which i can open in ADE, saving it as .epub but then I can still not transfer it to a new device which is how I found out about this method. 

Here is my log: 

calibre 5.28  embedded-python: True is64bit: False

Windows-10-10.0.19041 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '10', '10.0.19041')
Python 3.8.5
Windows: ('10', '10.0.19041', '', 'Multiprocessor Free')
Interface language: nl
Successfully initialized third party plugins: DeDRM (7, 2, 1)
calibre 5.28  embedded-python: True is64bit: False
Windows-10-10.0.19041 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '10', '10.0.19041')
Python 3.8.5
Windows: ('10', '10.0.19041', '', 'Multiprocessor Free')
Interface language: nl
Successfully initialized third party plugins: DeDRM (7, 2, 1)
Turning on automatic hidpi scaling
devicePixelRatio: 2.0
logicalDpi: 96.0 x 96.0
physicalDpi: 78.9126213592233 x 78.82758620689654
Using calibre Qt style: True
[0.00] Starting up...
[0.02] Showing splash screen...
[0.80] splash screen shown
[0.80] Initializing db...
[0.81] db initialized
[0.81] Constructing main UI...
Key 'B' for shortcut Browse annotations is already used by Boeken bijschaven, ignoring
[1.72] main UI initialized...
[1.72] Hiding splash screen
Starting QuickView
DeDRM v7.2.1: Trying to decrypt Bridgerton The Duke and I.epub
DeDRM v7.2.1: Verifying zip archive integrity
DeDRM v7.2.1: Bridgerton The Duke and I.epub is a secure Adobe Adept ePub
DeDRM v7.2.1: Looking for new default Adobe Digital Editions Keys after 0.2 seconds
Found 1 keys
DeDRM v7.2.1: Trying a new default key
Could not decrypt 7dc4cgoo.epub because of an exception:
Traceback (most recent call last):
  File "calibre_plugins.dedrm.ineptepub", line 418, in decryptBook
TypeError: object of type 'int' has no len()

DeDRM v7.2.1: Failed to decrypt with new default key after 0.5 seconds
DeDRM v7.2.1: Ultimately failed to decrypt after 0.5 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
[3.75] splash screen hidden
[3.75] Started up in 3.75 seconds with 21 books
stdout+stderr from file dialog helper: [b'', b'']
piped data from file dialog helper: [b'\x12h\xe6\xa8#\x08\x02\xcf\x9a\xef\xb2\x14\xf3\xd8\xec_\xaa\x89\xc9\x89^\xd4\r\x03\xc9`\xc7\xbd<\xa1v\xd3', b'C:\\Users\\irisv\\OneDrive\\Documenten\\Bridgerton The Duke and I.epub']
Added Bridgerton: The Duke and I to db in: 0.2
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "calibre\customize\ui.py", line 176, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 644, in run
  File "calibre_plugins.dedrm.__init__", line 415, in ePubDecrypt
calibre_plugins.dedrm.DeDRMError: DeDRM v7.2.1: Ultimately failed to decrypt after 0.5 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Exception ignored in: <function NativeFunction.__del__ at 0x077A2F58>
Traceback (most recent call last):
  File "calibre_plugins.dedrm.adobekey", line 268, in __del__
TypeError: 'NoneType' object is not callable
Exception ignored in: <function NativeFunction.__del__ at 0x077A2F58>
Traceback (most recent call last):
  File "calibre_plugins.dedrm.adobekey", line 268, in __del__
TypeError: 'NoneType' object is nAdded 1 books in 1.1 seconds 

Hopefully someone technical can help me out!

**Comment by ElleKayEm** [comment]
Adobe Digital Editions needs to be authorized with an Adobe ID before opening the acsm file.
